cocli/COMID.md

```diff
@@ -0,0 +1,129 @@
+
```

Comment:
If you still want to keep the split (which I don't think is necessary), at least move the new markdown files into a separate `doc` folder and adjust the links in the README.md accordingly.

Reply:
Not sure what the benefit for this suggestion is:

In fact it provides the facility for all the sub-README's to be in one palce (next to README). This is what we have followed in `github.com/veraison/docs/`

Reply:
The benefit is we don't pollute the cmd directory with random files.

Reply:
The README's are not in the `cmd` directory. It is in the `cocli` directory!

Reply:
Yes and it is fine to have all the README's in the `cocli` directory, which is the logical place for the same!

Reply:
> If you still want to keep the split (which I don't think is necessary) [...]

If you want to provide smoother navigation of lengthy documents, the usual approach is to create a table of contents.

Besides, sufficiently modern editors (including Visual Studio Code, which, if I remember correctly, is your IDE) can automatically create ToCs in markdown files.

On my laptop (also VSC-based), I get the following with a click:

- [Corim Command Line Interface](#corim-command-line-interface)
  - [Installing and configuring](#installing-and-configuring)
  - [CoMIDs manipulation](#comids-manipulation)
    - [Create](#create)
    - [Display](#display)
  - [CoTSs manipulation](#cotss-manipulation)
    - [Create](#create-1)
    - [Display](#display-1)
  - [CoRIMs manipulation](#corims-manipulation)
    - [Create](#create-2)
    - [Sign](#sign)
    - [Verify](#verify)
    - [Display](#display-2)
    - [Submit](#submit)
      - [Remote Service Authentication](#remote-service-authentication)
    - [Extract CoSWIDs, CoMIDs and CoTSs](#extract-coswids-comids-and-cotss)
  - [Visual Synopsis of the Available Commands](#visual-synopsis-of-the-available-commands)

Reply:
I see CoCLI documentation as a `book` with CoMID, CoTS and CoRIM (Creation, Display etc.) been separate chapters and 
hence on a separate `pages`.  I do not want to dump entire content on the same Page as it makes it very lengthy,  difficult to maintain in the future, especially with now `real world` examples ! Tomorrow new commands will be added making it even longer! That was the idea behind splitting the content.
We can still add a `ToC` with specific content on different pages, isn't it? I will explore that!

Reply:
> I see CoCLI documentation as a `book` with CoMID, CoTS and CoRIM (Creation, Display etc.) been separate chapters and hence on a separate `pages`.

the problem is it's not a book, it's just a readme - you're taking it too seriously ;)

> I do not want to dump entire content on the same Page as it makes it very lengthy, difficult to maintain in the future, especially with now `real world` examples !

It's not _very_ lengthy, it's just lengthy.

And no, it's not more difficult to maintain.

> Tomorrow new commands will be added making it even longer! That was the idea behind splitting the content.

There's no problem with that, it's just another line in the ToC - which, as I mentioned before, is just one click away in VSC.

> We can still add a `ToC` with specific content on different pages, isn't it? I will explore that!

That makes no sense.  It's not the encyclopedia that we are creating, it's just a readme file.

Reply:
All in one page the content is 500+ lines long which is `very lengthy` clearly.

Reply:
Well, see https://github.com/lestrrat-go/jwx/blob/develop/v2/docs/01-jwt.md